### ISSUE TITLE
Enhance accessibility to meet WCAG 2.0 compliance - automate compliance checking

### ISSUE THREAD

**Issue by jcohen02** [OxfordRSE/gutenberg#193]
The site should meet accessibility guidelines and be [WCAG 2.0](https://www.w3.org/TR/WCAG20/) compliant.

This will require some updates.

Compliance checking should then ideally be added into the GHA pipeline so that future changes that are non-compliant are flagged.

**Comment by mjaquiery** [comment]
We might want to use jest-axe in our component tests:
https://github.com/nickcolley/jest-axe?tab=readme-ov-file#testing-react-with-react-testing-library

**Comment by eatyourgreens** [comment]
WAVE is a very good tool for manually checking pages eg. https://wave.webaim.org/report#/https://train.oxrse.uk/material/HPCu/software_architecture_and_design/object_orientated/classes

**Comment by eatyourgreens** [comment]
Accessibility Insights is also a great tool for Chrome and Edge. It will walk you through a WCAG 2 audit, and integrates with GitHub so that you can open issues for any problems that you find.
https://accessibilityinsights.io/

Here's an example of an issue that it created:
https://github.com/zooniverse/front-end-monorepo/issues/1597

**Comment by eatyourgreens** [comment]
The Nomensa [Accessibility Statement Generator](https://www.accessibilitystatementgenerator.com) lists all the WCAG 2 AA success criteria. Tick/untick the ones that apply, and it will generate an accessibility statement that you can publish on the web site.

**Comment by eatyourgreens** [comment]
I ran a random course materials page through a Lighthouse check on Page Speed Insights and the results are pretty good, for Next.js. Performance is 69% and Accessibility scores 78%. This is without the PRs that merged yesterday, which should nudge the accessibility score up. 

https://pagespeed.web.dev/analysis/https-train-oxrse-uk-material-HPCu-introductory_courses-python-07_pandas_dataframes/xdjephxfsy?form_factor=desktop

**Comment by eatyourgreens** [comment]
I've started opening issues for WCAG failures. Should we have a label to group them too? Or maybe list them as sub-issues here?

**Comment by martinjrobins** [comment]
> I've started opening issues for WCAG failures. Should we have a label to group them too? Or maybe list them as sub-issues here?

label is good, thanks @eatyourgreens

**Comment by eatyourgreens** [comment]
Storybook 9 released today with improved WCAG testing built in (via `axe`, I think.) Here's a screenshot from another project where I'm using it.

<img width="900" alt="Automated accessibility testing in Storybook 9, showing a test failure for a React component that's missing keyboard control for a scrolling region." src="https://github.com/user-attachments/assets/4861b5ef-4a26-4eec-b119-7b895c7a1c33" />

Disclaimer: automated testing can only catch around ~30% of WCAG failures. Full WCAG compliance will require manual auditing of Gutenberg.
